File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/7_clase_de_23_agosto/clase_abstracta.dart
abstract class Animal{

  int patas = 0;
  String promedio_de_vida = "";
  String tipo_de_animal = "";
  String comida = "";
  void emitirsonido();

  void info();

}

class Perro implements Animal{

  int patas = 4;
  String promedio_de_vida = "10-13 años";
  String tipo_de_animal = "Mamifero";
  String comida = "carne";

  void emitirsonido(){
    print('Whuaauuuuuuu!!!');
  }

  void info(){
    print('El perro es un animal ${tipo_de_animal} de ${patas} patas, con un promedio de vida de ${promedio_de_vida} y su alimentacion es de ${comida}, su sonido caracteristico es:');
    emitirsonido();
  }

}

class Gato implements Animal{

  int patas = 4;
  String promedio_de_vida = "12-18 años";
  String tipo_de_animal = "Mamifero";
  String comida = "carne";

  void emitirsonido(){
    print('Miaauuuuuu!!!');
  }

  void info(){
    print('El gato es un animal ${tipo_de_animal} de ${patas} patas, con un promedio de vida de ${promedio_de_vida} y su alimentacion es de ${comida}, su sonido caracteristico es:');
    emitirsonido();
  }

}

void main(){

  Perro nuevo_perro = new Perro();
  nuevo_perro.info();

  Gato nuevo_gato = new Gato();
  nuevo_gato.info();

}

File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/7_clase_de_23_agosto/clase_abstracta.rb
class Animal

  attr_accessor :patas, :promedio_de_vida, :tipo_de_animal, :comida

  def emitirsonido()
    raise "porfavor implementa este metodo"
  end

  def info()
    raise "porfavor implementa este metodo"
  end

end

class Perro < Animal

  @@patas = 4
  @@promedio_de_vida = "10-13 años"
  @@tipo_de_animal = "Mamifero"
  @@comida = "carne"

  def emitirsonido()
    puts "Whuaauuuuuuu!!!"
  end

  def info()
    puts "El perro es un animal #{@@tipo_de_animal} de #{@@patas} patas, con un promedio de vida de #{@@promedio_de_vida} y su alimentacion es de #{@@comida}, su sonido caracteristico es:"
    emitirsonido()
  end

end

class Gato < Animal

  @@patas = 4
  @@promedio_de_vida = "12-18 años"
  @@tipo_de_animal = "Mamifero"
  @@comida = "carne"

  def emitirsonido()
    puts "Miaauuuuuu!!!"
  end

  def info()
    puts "El gato es un animal #{@@tipo_de_animal} de #{@@patas} patas, con un promedio de vida de #{@@promedio_de_vida} y su alimentacion es de #{@@comida}, su sonido caracteristico es:"
    emitirsonido()
  end

end


nuevo_perro = Perro.new()
nuevo_perro.info()

nuevo_gato = Gato.new()
nuevo_gato.info()